athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -217,18 +262,19 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
                     totalRecordCount = rs.getLong(1);
                 }
                 if (totalRecordCount > 0) {
-                    long pageCount = (long) (Math.ceil(totalRecordCount / MAX_PARTITION_COUNT));
-                    long partitionRecordCount = (totalRecordCount <= SINGLE_SPLIT_LIMIT_COUNT) ? (long) totalRecordCount : pageCount;
-                    LOGGER.info(" Total Page  Count" +  partitionRecordCount);
-                    double limit = (int) Math.ceil(totalRecordCount / partitionRecordCount);
+                    String primaryKey = getPrimaryKey(getTableLayoutRequest.getTableName());
+                    long recordsInPartition = (long) (Math.ceil(totalRecordCount / MAX_PARTITION_COUNT));
+                    long partitionRecordCount = (totalRecordCount <= SINGLE_SPLIT_LIMIT_COUNT || primaryKey.equals("")) ? (long) totalRecordCount : recordsInPartition;
```

Comment:
If returned Optional; this is easily checked for through optional if empty or exists. easier to read as well.